runtime/compiler/runtime/Listener.cpp

```diff
@@ -148,36 +148,26 @@ handleOpenSSLConnectionError(int connfd, SSL *&ssl, BIO *&bio, const char *errMs
       (*OBIO_free_all)(bio);
       bio = NULL;
       }
-   if (ssl)
-      {
-      (*OSSL_free)(ssl);
-      ssl = NULL;
-      }
    return false;
    }
 
 static bool
 acceptOpenSSLConnection(SSL_CTX *sslCtx, int connfd, BIO *&bio)
    {
-   SSL *ssl = (*OSSL_new)(sslCtx);
-   if (!ssl)
-      return handleOpenSSLConnectionError(connfd, ssl, bio, "Error creating SSL connection");
+   SSL *ssl = NULL;
+   bio = (*OBIO_new_ssl)(sslCtx, false);
+   if (!bio)
+      return handleOpenSSLConnectionError(connfd, ssl, bio, "Error creating new BIO");
 
-   (*OSSL_set_accept_state)(ssl);
```

Comment:
This statement no longer exists in the new code. Was that on purpose?

Reply:
yeah, because calling `SSL_accept` and `SSL_connect` already sets the appropriate state.
https://www.openssl.org/docs/man1.1.0/man3/SSL_set_accept_state.html